### PR TITLE
migrate code from googleapis/python-vision

### PR DESCRIPTION
**Pull request by arbrown**
- chore: move samples from python-docs-sample (#45)
- feat!: migrate to use microgen (#52)
- chore(deps): update dependency google-cloud-vision to v2 (#55)
- chore(deps): update dependency pillow to v8 (#65)
- chore(deps): update dependency google-cloud-storage to v1.31.2 (#50)
- chore(deps): update dependency pillow to v8.0.1 (#71)
- chore(deps): update dependency google-cloud-storage to v1.32.0 (#69)
- chore(deps): update dependency google-cloud-storage to v1.33.0 (#75)
- chore: fix flaky test by adding rerun (#77)
- chore(deps): update dependency google-cloud-storage to v1.35.0 (#80)
- feat(v1): add text detection params; fix: remove client side recv limits (#82)
- chore(deps): update dependency pillow to v8.1.0 (#90)
- chore(deps): update dependency google-cloud-vision to v2.1.0 (#99)
- test: add 3.9 tests (#96)
- chore(deps): update dependency google-cloud-vision to v2.2.0 (#109)
- chore(deps): update dependency google-cloud-storage to v1.36.0 (#101)
- chore(deps): update dependency google-cloud-storage to v1.36.1 (#113)
- chore(deps): update dependency pillow to v8.1.1 (#114)
- chore(deps): update dependency pillow to v8.1.2 (#115)
- chore(deps): update dependency google-cloud-storage to v1.36.2 (#116)
- chore(deps): update dependency google-cloud-storage to v1.37.0 (#121)
- chore(deps): update dependency google-cloud-vision to v2.3.0 (#124)
- chore(deps): update dependency pillow to v8.2.0 (#126)
- chore(deps): update dependency google-cloud-vision to v2.3.1 (#135)
- chore: migrate to owl bot (#143)
- chore(deps): update dependency pytest to v6.2.4 (#144)
- chore(deps): update dependency google-cloud-storage to v1.38.0 (#127)
- chore: new owl bot post processor docker image (#151)
- chore: increased timeout by 200s to reduce flakiness (#155)
- chore(deps): update dependency google-cloud-vision to v2.3.2 (#160)
- chore(deps): update dependency google-cloud-storage to v1.39.0 (#167)
- chore(deps): update dependency pillow to v8.3.0 (#173)
- chore(deps): update dependency google-cloud-storage to v1.40.0 (#172)
- chore(deps): update dependency pillow to v8.3.1 (#174)
- chore(deps): update dependency backoff to v1.11.0 (#177)
- chore(deps): update dependency backoff to v1.11.1 (#196)
- chore(deps): update dependency google-cloud-storage to v1.41.0 (#195)
- chore(deps): update dependency google-cloud-vision to v2.4.0 (#178)
- chore(deps): update dependency google-cloud-storage to v1.41.1 (#203)
- feat: add Samples section to CONTRIBUTING.rst (#201)
- chore(deps): update dependency google-cloud-vision to v2.4.1 (#204)
- chore(deps): update dependency google-cloud-vision to v2.4.2 (#211)
- chore: fix INSTALL_LIBRARY_FROM_SOURCE in noxfile.py (#213)
- chore: drop mention of Python 2.7 from templates (#217)
- chore(deps): update dependency google-cloud-storage to v1.42.0 (#215)
- chore(deps): update dependency pytest to v6.2.5 (#222)
- build(deps): bump pillow in /samples/snippets/crop_hints (#228)
- chore(deps): update dependency google-cloud-storage to v1.42.1 (#229)
- chore(deps): update dependency pillow to v8.3.2 (#230)
- chore(deps): update dependency google-cloud-storage to v1.42.2 (#232)
- chore: blacken samples noxfile template (#231)
- chore(deps): update dependency google-cloud-vision to v2.4.3 (#238)
- chore(deps): update dependency google-cloud-vision to v2.4.4 (#242)
- chore(deps): update dependency google-cloud-storage to v1.42.3 (#243)
- chore: fail samples nox session if python version is missing (#241)
- chore(python): Add kokoro configs for python 3.10 samples testing (#250)
- chore(deps): update dependency google-cloud-vision to v2.5.0 (#251)
- chore(deps): update dependency pillow to v8.4.0 (#255)
- chore(deps): update dependency google-cloud-vision to v2.6.0 (#258)
- docs: update async_detect_document() sample (#260)
- chore(deps): update dependency google-cloud-vision to v2.6.1 (#262)
- chore(deps): update dependency google-cloud-vision to v2.6.2 (#265)
- chore(deps): update dependency google-cloud-storage to v1.43.0 (#274)
- docs: Update doctext sample to include method signature (#275)
- docs: Add example of how to use max_results (#277)
- chore: update python-docs-samples link to main branch (#278)
- chore(deps): update dependency google-cloud-vision to v2.6.3 (#281)
- chore(samples): Add check for tests in directory (#288)
- chore(deps): update all dependencies (#285)
- chore(deps): update dependency google-cloud-storage to v2 (#294)
- chore(python): Noxfile recognizes that tests can live in a folder (#296)
- chore(deps): update dependency google-cloud-storage to v2.1.0 (#297)
- chore(deps): update all dependencies (#311)
- chore(deps): update all dependencies (#313)
- chore: Adding support for pytest-xdist and pytest-parallel (#320)
- chore(deps): update dependency google-cloud-vision to v2.7.1 (#323)
- chore(deps): update dependency pytest to v7.1.0 (#325)
- chore(deps): update dependency google-cloud-storage to v2.2.0 (#326)
- chore(deps): update dependency google-cloud-storage to v2.2.1 (#327)
- chore(deps): update dependency pytest to v7.1.1 (#328)
- chore(python): use black==22.3.0 (#333)
- chore(deps): update dependency google-cloud-vision to v2.7.2 (#332)
- chore(deps): update dependency pillow to v9.1.0 (#338)
- chore(deps): update dependency google-cloud-storage to v2.3.0 (#345)
- chore(python): add nox session to sort python imports (#347)
- chore(deps): update dependency pytest to v7.1.2 (#350)
- chore(deps): update dependency backoff to v2 (#351)
- chore(deps): update dependency backoff to v2.0.1 (#353)
- chore(deps): update dependency pillow to v9.1.1 (#359)
- fix: require python 3.7+ (#452)
- chore(deps): update all dependencies (#444)
- chore(deps): update all dependencies (#461)
- chore(deps): update all dependencies (#463)
- chore(deps): update all dependencies (#466)
- chore(deps): update dependency google-cloud-vision to v3.1.1 (#471)
- chore(deps): update dependency pytest to v7.1.3 (#480)
- chore: detect samples tests in nested directories (#484)
- chore(deps): update dependency google-cloud-vision to v3.1.2 (#488)
- chore(deps): update dependency google-cloud-vision to v3.1.3 (#492)
- chore(deps): update dependency backoff to v2.2.1 (#493)
- chore(deps): update dependency google-cloud-vision to v3.1.4 (#496)
- chore(deps): update dependency pytest to v7.2.0 (#497)
- chore(deps): update dependency pillow to v9.3.0 (#499)

## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: vision/snippets/crop_hints/noxfile.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
+# Copyright 2019 Google LLC
```
Comment (dandhlee) [inline vision/snippets/crop_hints/noxfile.py:1]: Please remove this file.

File: vision/snippets/crop_hints/noxfile_config.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# You can copy this file into your directory, then it will be inported from
+# the noxfile.py.
+
+# The source of truth:
+# https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/noxfile_config.py
+
+TEST_CONFIG_OVERRIDE = {
+    # You can opt out from the test for specific Python versions.
+    # Pillow 9.0.0 does not support python 3.6
+    "ignored_versions": ["3.6"],
```
Comment (dandhlee) [inline vision/snippets/crop_hints/noxfile_config.py:26]:
```suggestion
    "ignored_versions": ["2.7", "3.6"],
```

File: vision/snippets/detect/noxfile.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
+# Copyright 2019 Google LLC
```
Comment (dandhlee) [inline vision/snippets/detect/noxfile.py:1]: Please remove.

File: vision/snippets/document_text/noxfile_config.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# You can copy this file into your directory, then it will be inported from
+# the noxfile.py.
+
+# The source of truth:
+# https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/noxfile_config.py
+
+TEST_CONFIG_OVERRIDE = {
+    # You can opt out from the test for specific Python versions.
+    # Pillow 9.0.0 does not support python 3.6
+    "ignored_versions": ["3.6"],
```
Comment (dandhlee) [inline vision/snippets/document_text/noxfile_config.py:26]:
```suggestion
    "ignored_versions": ["2.7", "3.6"],
```

File: vision/snippets/face_detection/noxfile.py [hunk 5]
```diff
@@ -0,0 +1,312 @@
+# Copyright 2019 Google LLC
```
Comment (dandhlee) [inline vision/snippets/face_detection/noxfile.py:1]: Please remove.

File: vision/snippets/face_detection/noxfile_config.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+# You can copy this file into your directory, then it will be inported from
+# the noxfile.py.
+
+# The source of truth:
+# https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/noxfile_config.py
+
+TEST_CONFIG_OVERRIDE = {
+    # You can opt out from the test for specific Python versions.
+    # Pillow 9.0.0 does not support python 3.6
+    "ignored_versions": ["3.6"],
```
Comment (dandhlee) [inline vision/snippets/face_detection/noxfile_config.py:26]:
```suggestion
    "ignored_versions": ["2.7", "3.6"],
```

File: vision/snippets/quickstart/noxfile.py [hunk 7]
```diff
@@ -0,0 +1,312 @@
+# Copyright 2019 Google LLC
```
Comment (dandhlee) [inline vision/snippets/quickstart/noxfile.py:1]: Please remove.

File: vision/snippets/web/noxfile.py [hunk 8]
```diff
@@ -0,0 +1,312 @@
+# Copyright 2019 Google LLC
```
Comment (dandhlee) [inline vision/snippets/web/noxfile.py:1]: Please remove.

**Review by dandhlee — CHANGES_REQUESTED** [review]
Could you add `CODEOWNERS` and `blunderbuss` for `api: vision`?

File: vision/snippets/crop_hints/noxfile.py (hunk 1 above)
Comment (arbrown) [reply vision/snippets/crop_hints/noxfile.py:1]: Done.

File: .github/CODEOWNERS [hunk 9]
```diff
@@ -69,5 +69,6 @@
 /trace/**/*                            @ymotongpoo @GoogleCloudPlatform/python-samples-reviewers
 /translate/**/*                        @nicain @GoogleCloudPlatform/python-samples-reviewers
 /talent/**/*                           @GoogleCloudPlatform/python-samples-reviewers
+/vision/**/*                           @googleapis/python-samples-reviewers @googleapis/cdpe-cloudai
```
Comment (dandhlee) [inline .github/CODEOWNERS:72]: It needs to be a team in the `GoogleCloudPlatform` org. If you're not sure of the correct one, we can leave it defaulted to `python-sample-reviewers` for now.

File: .github/blunderbuss.yml [hunk 10]
```diff
@@ -131,6 +131,10 @@ assign_issues_by:
   - 'api: cloudkms'
   to: 
   - GoogleCloudPlatform/dee-infra
+- labels:
+  - 'api: vision'
+  to:
+  - googleapis/cdpe-cloudai
```
Comment (dandhlee) [inline .github/blunderbuss.yml:137]: Ditto.

**Review by dandhlee — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 9 above)
Comment (arbrown) [reply .github/CODEOWNERS]: Ok, done.  That's what it was in the old repo, and I couldn't find the same team in the GoogleCloudPlatform org.

File: .github/CODEOWNERS [hunk 11]
```diff
@@ -69,5 +69,6 @@
 /trace/**/*                            @ymotongpoo @GoogleCloudPlatform/python-samples-reviewers
 /translate/**/*                        @nicain @GoogleCloudPlatform/python-samples-reviewers
 /talent/**/*                           @GoogleCloudPlatform/python-samples-reviewers
+/vision/**/*                           @googleapis/python-samples-reviewers
```
Comment (dandhlee) [inline .github/CODEOWNERS:72]:
```suggestion
/vision/**/*                           @GoogleCloudPlatform/python-samples-reviewers
```

File: vision/snippets/crop_hints/noxfile.py (hunk 1 above)
Comment (dandhlee) [reply vision/snippets/crop_hints/noxfile.py:1]: Missed this file 😓

**Review by dandhlee — APPROVED** [review]
LGTM, overriding CLA and merging.

**Pull request merged**